### ISSUE TITLE
Add new tools for Mars Navigator

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#1388]
**Is your feature request related to a problem? Please describe.**
- When it comes to mission planning for expeditions, Mars Navigator should be the go-to toolbox for players.
- Currently, this tool only has a zoom slider, a button for selecting a surface map and the option of turning on and off a myriads of map layers. 
- We propose the following tools to be made available 
   - A ruler for measuring distances between any two points 
   - A tracer tool for creating topo lines (not straight lines)
   - A area measurement (square km) tool for estimating the the mineral reserve volume or mass  
- Some of the capabilities would be shared with the surface map in Mission Tool's Nav tab, which is also based on Mars Navigator.

**Describe the rationale**
- Mars Navigator can be more feature rich for visual mission planning. 

**Describe the initial goal(s) you have in mind**
- Drop custom markers
- Trace topo lines between two markers
- Measuring distances (along a selected topo line or a straight line) between any two markers 

**Describe the non-goal(s) associated with the goal(s)**
- In future, rework the switching of surface map's resolution level to be more player-friendly
- Allow two maps to be simultaneously loaded side-by-side for visual comparison

**Describe the solution you desire to see**
- Add mouse controls for defining a start point and an end point
- Add algorithms for tracing topo lines (lowest possible change of adjacent color pixels) 

**Specifications**
- post-3.8.0

**Comment by mokun** [comment]
@bevans2000 

Btw, I found out not all maps are centered (say at Latitude 0 and longitude 0) at the same location. They can be a little bit off.

Thus a settlement may be shown inside a small crater in one surface map but outside of that same crater on another surface map. 

I may need to pull back the newly added *Vertical Roughness* map.

**Comment by mokun** [comment]
@bevans2000 @muybien3433 

I'm relocating the zoom slider bar that's on the right hand side of Mars Navigator.

It has been attached to Mars Navigator. I'll make it float on top of the map and on the right hand side of the map, much like how it appears on the Settlement Map. 

This way, the nav tab map may be given the capability of the zoom slider as well.

**Comment by bevans2000** [comment]
@mokun are you finished with you changes to the Map Tool for a short period? I want to continue with #1037 but will now need to make changes to the Map tool itself

**Comment by mokun** [comment]
> are you finished with you changes to the Map Tool for a short period? 

Sure. I'll pause.

**Comment by bevans2000** [comment]
> > are you finished with you changes to the Map Tool for a short period?
> 
> Sure. I'll pause.

Cheers. I'm nearly there

**Comment by mokun** [comment]
Can you by any chance leave the mechanism only loosely coupled with maps ? 

I'd like to extend this ability for downloading not just maps, but also the elevation data sets, as well as sound track files. 

FYI @muybien3433

**Comment by bevans2000** [comment]
Yes, file async downloading, ```FileLocator```, is done on the branch. Just have to get the map display to accommodate it.

**Comment by mokun** [comment]
Awesome !

Q: how would you prefer to select the resolution level of a map ? 

Currently, it's a little awkward. 

Player has to reselect the same map type in order for a dialog box to pop up. 

Then player may choose a different resolution level, which will trigger opening a map that provides a different level of detail.

We could make it a bit more intuitive and easy.

**Comment by mokun** [comment]
Having said that, in future, after we have implemented tile loading, we may not even need to get the player to explicitly choose a resolution level.

It's simply that as player roll the mouse wheel up or down, it will automatically go up and down the resolution.

**Comment by bevans2000** [comment]
Yes I agree, the resolution selection should be automatic. Let's pick that up later.

**Comment by mokun** [comment]
@bevans2000 

I'm interested in adding a color picker/palette for player to choose a custom color for landmarks/, settlements, or vehicles.

But I'll wait for you to complete the map layer optimization first.

**Comment by bevans2000** [comment]
I had a thought about this. I think we should change the logic so any map layer could support filtering not just the mineral layer.

**Comment by bevans2000** [comment]
There are 2 changes that this should include

- The control should be moved from the South to the East. The status bar is not always visible height is larger than the desktop height. Controls need laying out vertically
- A scale should be shown in the map for KMs per pixel or a scale range

**Comment by mokun** [comment]
@bevans2000 

When I open Mars Navigator after 2 or 3 sols, I often found the settlement names being overlapped with mission vehicle names. See below : 

<img width="758" height="609" alt="Image" src="https://github.com/user-attachments/assets/87a9f288-72f9-4528-9b12-c62c72c41184" />

I know all I need to do is to filter out the vehicle names and let settlement names stay.

Currently we couldn't do it.

In UnitMapLayer,
```java
	@Override
	public List<MapFilter> getFilterDetails() {
		List<MapFilter> filters = new ArrayList<>();
		filters.add(new MapFilter(LABEL_FILTER, "Labels", labelIcon));
		return filters;
	}

	@Override
	public void displayFilter(String name, boolean display) {
		if (name.equals(LABEL_FILTER)) {
			displayLabel = display;
		}
	}

	@Override
	public boolean isFilterActive(String filterName) {
		return (LABEL_FILTER.equals(filterName) && displayLabel);
	}
```

I have the impression that we should be under the checkbox box called 'Units', we need `Settlement Labels`, `Vehicle Labels`, etc.

What would you add to the codes to make it possible ?

**Comment by bevans2000** [comment]
I think that should be possible. I'll to have a look

**Comment by mokun** [comment]
Currently, I'm on a dark theme. It's difficult to tell which map has been loaded in Mars Navigator. Also difficult to see which map have been downloaded. See pic below.

<img width="588" height="746" alt="Image" src="https://github.com/user-attachments/assets/382cdc4f-3e03-4431-b14d-56af86031291" />

In comparison, if I'm in a light theme, it's easy. See pic below.

<img width="551" height="713" alt="Image" src="https://github.com/user-attachments/assets/e747c76c-12f2-486c-a159-e5a4dbb0400e" />

So how do we override the LAF to get a lighter color for both the selected icon and downloaded icon ?

**Comment by bevans2000** [comment]
Not sure. Where does the eye icon come from? Looks like it adjusts the the LAF already so maybe it's just using the wrong colour in the style.

**Comment by bevans2000** [comment]
Actually thinking about it, you just have to make sure the icon is using the text foreground colour.

**Comment by mokun** [comment]
As you can see, the "eye icon" is in black, while at the same time, I'm using Night Owl Dark Theme. They don't have a good contrast.

It's quite difficult to see that icon that's right in front of the "Viking Colorized Global Mosaic..." surface map since my mouse cursor is on top of another map ("Marine 9 Geology").

<img width="399" height="303" alt="Image" src="https://github.com/user-attachments/assets/873bf1d2-aab2-4d87-9eaf-d67b1db0b14a" />

If I move the cursor on top of that Viking map, the contrast become a bit better and I can see it.

<img width="396" height="355" alt="Image" src="https://github.com/user-attachments/assets/fbe5f9a2-24d8-42da-b580-4f6626e942a4" />

**Comment by mokun** [comment]
@bevans2000 

I just added 2 checkboxes nodes, namely Settlement(s) and Vehicle(s), under "Units". 

Eventually, I want to use non-black colors text for Settlement(s) and Vehicle(s) separately.

And I give the player the flexibility to change color.

If a surface map has a dull color background and in that case, the settlement/vehicle text color need to be bright.

But if a surface map has a bright color background, then the player can switch to a black or dark color settlement/vehicle text color.

How do I do that ?

**Comment by bevans2000** [comment]
There is a property in the MapMetaData class that states if the map is colorful. That property drives the colours used for labels. 
One option to solve your problem would be to define the label colour directly in the metadata but that would be effort to quantify what colour to use
